mlir/lib/Conversion/MIGraphXToTosa/MIGraphXToTosa.cpp

```diff
@@ -235,9 +235,38 @@ LogicalResult ConvConverter<ConvType>::matchAndRewrite(
   newShape.push_back(outShape[1]);
   Type newOutTy = RankedTensorType::get(newShape, outputTy.getElementType());
 
+  auto expandTo2D = [&rewriter, loc](mlir::Value value) {
+    ArrayRef<int64_t> origShape = value.getType().cast<ShapedType>().getShape();
+    SmallVector<int64_t> expShape(origShape.drop_back());
+    expShape.push_back(1);
+    expShape.push_back(origShape.back());
+    Value reshaped = rewriter.create<tosa::ReshapeOp>(
+        loc, value, rewriter.getDenseI64ArrayAttr(expShape));
+    return reshaped;
+  };
+
   // Construct a new Conv2DOp.
   Operation *cop;
+  Type new1DOutTy;
   switch (dims) {
+  case 1:
+    // Expand to do a conv2d, because there's no conv1d op.
+    newShape.insert(std::prev(newShape.end()), 1);
+    new1DOutTy = RankedTensorType::get(newShape, outputTy.getElementType());
+    input = expandTo2D(input);
+    filter = expandTo2D(filter);
+
+    cop = rewriter.create<tosa::Conv2DOp>(
+        loc, new1DOutTy,
+        ValueRange{
+            input, filter,
+            getZeroTensor(
+                loc, outputTy.getElementType(),
+                filter.getType().template cast<ShapedType>().getShape()[0],
+                rewriter)});
+    cop->setAttr("rock.expanded_from_1d", rewriter.getBoolAttr(true));
```

Comment:
please add this to RockAttrDefs and use here.

Reply:
Thanks, I didn't know about that.